WalletWasabi.Gui/CrashReport/CrashReporter.cs

```diff
@@ -31,8 +31,10 @@ public void TryInvokeCrashReport()
 
 				var args = $"crashreport -attempt=\"{Attempts + 1}\" -exception=\"{Base64ExceptionString}\"";
 
-				ProcessBridge processBridge = new ProcessBridge(Process.GetCurrentProcess().MainModule.FileName);
-				processBridge.Start(args, false);
+				ProcessStartInfo startInfo = ProcessStartInfoFactory.Make(Process.GetCurrentProcess().MainModule.FileName, args);
+				using Process p = Process.Start(startInfo);
+				p.Start();
+				p.WaitForExit();
```

Comment:
My issue with this is blocking.

Reply:
> Why is this necessary? Could it accept a max. time to wait?

When `ProcessBridge` instantiates a new `System.Diagnostics.Process`, it does handle dispose logic later on. So I think, in this case, it leaks unmanaged memory (https://stackoverflow.com/questions/16957320/what-does-process-dispose-actually-do).

What would maximum time would be good for? 

> My issue with this is blocking.

Given that this `CrashReporter` is invoked very late in DisposeAsync part of the WalletWasabi/Program.cs, I'm not sure what the issue is here. From highlevel point of view, it should be something like:

* WasabiWallet.GUI stopped everything.
* Open CrashReporter (currently it is in modal style)
* Wait until user closes CrashReporter
* Stop everything.

Not sure, whether there is better way to do it.

Reply:
If this is automatic handling of dispose logic then it is not really necessary, as the referenced SO answer says, when an instance goes out of scope it is garbage collected and the Dispose method is called automatically releasing all the unmanaged resources. Dispose is used for releasing resources as soon as possible but they are released anyway by dotnet. 

Anyway, that's not the point here, the point is that blocking a thread without any warranty of being eventually unblocked seems dangerous.

Reply:
@lontivero It is true that it will release resources later on. However, suppose we have this:

```csharp
  ProcessBridge processBridge = new ProcessBridge(Process.GetCurrentProcess().MainModule.FileName);
  ProcessStartInfo startInfo = ProcessStartInfoFactory.Make(Process.GetCurrentProcess().MainModule.FileName, args);
  processBridge.Start(args, false);
} // at this point, `processBridge` is eligible for garbage collection, suppose that the garbage collection happens right at this moment. What does happen? Given that process Dispose() method is called. The process is closed: https://source.dot.net/#System.Diagnostics.Process/System/Diagnostics/Process.cs,802 thanks to Dispose method.
```

So my understanding is that, it may simply happen that you start the application and it may stop at any moment when garbage collection happen. I have not seen that happen but I can't happen to persuade myself that it can't happen.

Reply:
Ok, so Lucas is right. The blocking is not necessary and the method `Process.Close()` does not do what one may think. :-) 

A few tests:

```csharp
using System;
using System.Diagnostics;

namespace DisposeTest
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("#1");
            Test01(); // Does not close the program.

            Console.WriteLine("#2");
            Test02(); // Does not close the program.

            Console.WriteLine("#3");
            Test03(); // Does not close the program.

            // https://stackoverflow.com/questions/673031/process-close-is-not-terminating-created-process-c
            Console.WriteLine("#4");
            Test04(); // Does kill the program.

            Console.WriteLine("#5");
        }

        public static void Test01()
        {
            using Process process = Process.Start("notepad.exe");
        }

        public static void Test02()
        {
            Process process = Process.Start("notepad.exe");
            process.Dispose();
        }
        public static void Test03()
        {
            // https://stackoverflow.com/questions/673031/process-close-is-not-terminating-created-process-c
            using Process process = Process.Start("notepad.exe");
            process.Close();
        }

        public static void Test04()
        {
            using Process process = Process.Start("notepad.exe");
            process.Kill();
        }
    }
}
```